kasa/tests/test_discovery.py

```diff
@@ -170,12 +170,13 @@ async def test_discover_single_hostname(discovery_mock, mocker):
 async def test_discover_credentials(mocker):
     """Make sure that discover gives credentials precedence over un and pw."""
     host = "127.0.0.1"
-    mocker.patch("kasa.discover._DiscoverProtocol.wait_for_discovery_to_complete")
 
-    def mock_discover(self, *_, **__):
+    async def mock_discover(self, *_, **__):
         self.discovered_devices = {host: MagicMock()}
+        self.seen_hosts.add(host)
+        self._handle_discovered_event()
 
-    mocker.patch.object(_DiscoverProtocol, "do_discover", mock_discover)
+    mocker.patch.object(_DiscoverProtocol, "do_discover", new=mock_discover)
```

Comment:
These tests were slightly erroneous before as "do_discover" has been async for some time, but they would have completed before the physical connection called back on `connection_made` and tried to start the `do_discover` task. As the fixture now calls back `connection_made` immediately, the error reveals itself.

Reply:
Nice catch :+1: